application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -1650,7 +1650,19 @@ public function get_participant_properties($sSessionKey, $iSurveyID, $iTokenID,
                 if(!tableExists("{{tokens_$iSurveyID}}"))
                     return array('status' => 'Error: No token table');
 
-                $token = Token::model($iSurveyID)->findByPk($iTokenID);
+                if(is_array($aTokenQueryProperties)){
+		    $tokens = Token::model($iSurveyID)->findAllByAttributes($aTokenQueryProperties);
```

Comment:
To optimize and not pull a HUGE result case (imagine the table has 100,000 tokens) you probably should use ->count() here and if it is 1 do the findByAttributes...